README.md

```diff
@@ -36,110 +36,41 @@ crossbeam-channel = "0.4.0"
 notify = "5.0.0-pre.14"
 ```
 
-## Usage
+## Usage (5.0.0-pre.14)
 
 The examples below are aspirational only, to preview what the final release may
 have looked like. They may not work. Refer to [the API documentation][docs] instead.
 
 ```rust
-use notify::{RecommendedWatcher, RecursiveMode, Result, watcher};
-use std::time::Duration;
+use notify::{RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::Path;
+
+fn main() -> notify::Result<()> {
+    let (tx, rx) = std::sync::mpsc::channel();
 
-fn main() -> Result<()> {
     // Automatically select the best implementation for your platform.
     // You can also access each implementation directly e.g. INotifyWatcher.
-    let mut watcher = watcher(Duration::from_secs(2))?;
+    // Note that dropping this `watcher` will exit the Watcher-Loop itself.
+    let mut watcher = RecommendedWatcher::new(tx)?;
 
     // Add a path to be watched. All files and directories at that path and
     // below will be monitored for changes.
-    watcher.watch("/home/test/notify", RecursiveMode::Recursive)?;
-
-    // This is a simple loop, but you may want to use more complex logic here,
-    // for example to handle I/O.
-    for event in &watcher {
-        match event {
-            Ok(event) => println!("changed: {:?}", event.path),
-            Err(err) => println!("watch error: {:?}", err),
-        };
-    }
-
-    Ok(())
-}
-```
-
-### With a channel
-
-To get a channel for advanced or flexible cases, use:
-
-```rust
-let rx = watcher.channel();
+    watcher.watch(Path::new("."), RecursiveMode::Recursive)?;
 
-loop {
-    match rx.recv() {
-        // ...
+    for res in rx {
+        match res {
+            Ok(event) => println!("changed: {:?}", event),
+            Err(e) => println!("watch error: {:?}", e),
+        }
     }
-}
-```
-
-To pass in a channel manually:
-
-```rust
-let (tx, rx) = crossbeam_channel::unbounded();
-let mut watcher: RecommendedWatcher = Watcher::with_channel(tx, Duration::from_secs(2))?;
 
-for event in rx.iter() {
-    // ...
+    Ok(())
 }
 ```
 
-### With precise events
```

Comment:
We do still talk about that in lib.rs. It's a planned feature but for now completely wrong. As we always emit precise events.